### DISCUSSION TITLE
Shields.io badges don't work on AutomationHub

### DISCUSSION THREAD

**Discussion by andy-maier** [ansible-collections/overview#202] [Q&A] [answered]
Hello, I'd like to show a badge with the latest version of our collection on the README page of its GitHub repo.

I have this working for Ansible Galaxy using the badges from shields.io:

https://img.shields.io/badge/dynamic/json?style=flat&label=galaxy&prefix=v&url=https://galaxy.ansible.com/api/v2/collections/ibm/ibm_zhmc/&query=latest_version.version

Shields.io presents a badge with "galaxy | v1.0.2" based on the JSON returned by a GET to this URL:

https://galaxy.ansible.com/api/v2/collections/ibm/ibm_zhmc/

For Ansible AutomationHub, I found a similar query that returns the latest version:

https://console.redhat.com/api/automation-hub/v3/collections/ibm/ibm_zhmc/

But when I use this for shields.io to define the badge, the badge shows "hub | invalid":

https://img.shields.io/badge/dynamic/json?style=flat&label=hub&prefix=v&url=https://console.redhat.com/api/automation-hub/v3/collections/ibm/ibm_zhmc/&query=highest_version.version

I suspected this has to do with either blocking the traffic or missing authentication, and after a few attempts to get rid of any auth context of my browser, I could verify that it is an authentication issue. When you are sufficiently logged out from Hub, the URL above results in "Missing Authentication" being shown by the browser, so supposedly a HTTP 403.

Is there a chance that AutomationHub allows the version query without being authenticated?

Are there alternative ways to create version badges?

**Comment by gundalow** [comment, marked as answer]
Seems like you are right, if you visit https://console.redhat.com/api/automation-hub/v3/collections/ibm/ibm_zhmc/ without being logged in (ie in incognito mode) you get `Missing Authentication`.

This should get resolved as part of the work to move galaxy.ansible.com to the GalaxyNG codebase (which powers Automation Hub). Though realistically that's 18+ months away.

I don't know of any way to work around this. 

I've asked the Automation Hub development team.

**Reply by andy-maier** [reply]
@gundalow Hi John. Thanks for clarifying this. We'll remove the AutomationHub badge for now to let the front page look better, and if any new info comes out of the dev team, please let me know.

**Comment by andy-maier** [comment]
Just for the record: As of today, it still returns missing authentication when not logged in: https://console.redhat.com/api/automation-hub/v3/collections/ibm/ibm_zhmc/
